includes/admin/meta-boxes/class-wcs-meta-box-related-orders.php

```diff
@@ -21,47 +20,47 @@ class WCS_Meta_Box_Related_Orders {
 
 	/**
 	 * Output the metabox
+	 * @param  WP_Post|WC_Order $post_or_order_object The post object or order object currently being edited.
 	 */
-	public static function output( $post ) {
-
-		if ( wcs_is_subscription( $post->ID ) ) {
-			$subscription = wcs_get_subscription( $post->ID );
-			$order = ( false == $subscription->get_parent_id() ) ? $subscription : $subscription->get_parent();
-		} else {
-			$order = wc_get_order( $post->ID );
-		}
+	public static function output( $post_or_order_object ) {
+		$order = ( $post_or_order_object instanceof WP_Post ) ? wc_get_order( $post_or_order_object->ID ) : $post_or_order_object;
 
 		add_action( 'woocommerce_subscriptions_related_orders_meta_box_rows', __CLASS__ . '::output_rows', 10 );
 
-		include_once( dirname( __FILE__ ) . '/views/html-related-orders-table.php' );
+		include_once dirname( __FILE__ ) . '/views/html-related-orders-table.php';
 
-		do_action( 'woocommerce_subscriptions_related_orders_meta_box', $order, $post );
+		do_action( 'woocommerce_subscriptions_related_orders_meta_box', $order );
 	}
 
 	/**
 	 * Displays the renewal orders in the Related Orders meta box.
 	 *
-	 * @param object $post A WordPress post
+	 * @param WC_Order|WC_Subscription $order The order or subscription object being used to display the related orders.
 	 * @since 1.0.0 - Migrated from WooCommerce Subscriptions v2.0
 	 */
-	public static function output_rows( $post ) {
-		$orders_to_display      = array();
-		$subscriptions          = array();
-		$initial_subscriptions  = array();
-		$orders_by_type         = array();
-		$unknown_orders         = array(); // Orders which couldn't be loaded.
+	public static function output_rows( $order ) {
+		$orders_to_display     = array();
+		$subscriptions         = array();
+		$initial_subscriptions = array();
+		$orders_by_type        = array();
+		$unknown_orders        = array(); // Orders which couldn't be loaded.
+		$is_subscription       = wcs_is_subscription( $order );
+		$this_order_id         = $order->get_id();
 
 		// If this is a subscriptions screen,
-		if ( wcs_is_subscription( $post->ID ) ) {
-			$this_subscription = wcs_get_subscription( $post->ID );
+		if ( $is_subscription ) {
+			$subscription      = wcs_get_subscription( $order );
+			$this_subscription = wcs_get_subscription( $order->get_id() );
```

Comment:
These two lines do the same thing right?

Reply:
Thanks, resolved in cd5e649